src/profiler/safe.rs

```diff
@@ -0,0 +1,11 @@
+use super::{result, sys};
+
+pub use result::DriverError;
+
+pub fn start() -> Result<(), DriverError> {
+    unsafe { sys::cuProfilerStart() }.result()
+}
+
+pub fn stop() -> Result<(), DriverError> {
+    unsafe { sys::cuProfilerStop() }.result()
+}
```

Comment:
It seems like these could be moved into the driver module instead of creating a new profiler module?

And then after that lets rename to start_profiler() and stop_profiler()

Reply:
Well this modifies `wrapper.h`. Since you would need to include `cudaProfiler.h`.

I was hesitant to do it that way, but I could definitely do it.